common/dev/guid.c

```diff
@@ -20,8 +22,11 @@ uint8_t GUID_read(EEPROM_ENTRY *entry)
 		return GUID_OUT_OF_RANGE;
 	}
 
-	memcpy(&entry->config, &guid_config[entry->config.dev_id],
-	       sizeof(guid_config[entry->config.dev_id]));
+	if (entry->config.dev_id >= (sizeof(guid_config) / sizeof(EEPROM_CFG))) {
+		return GUID_OUT_OF_RANGE;
+	}
+
+	memcpy(&entry->config, &guid_config[entry->config.dev_id], sizeof(EEPROM_CFG));
```

Comment:
We should check that entry != NULL when passed in

Reply:
The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .

Reply:
Hi @GoldenBug 
The optimization code is based on v2022.07.01. After that, the main branch has been phased in some
commits related to sensor monitoring. Those changes will be phased in main_v2 in the other PRs. This
problem will be solved in those PRs.
Please check #223  .